src/main/java/Duke.java

```diff
@@ -1,10 +1,224 @@
+                System.out.println(line);
+                System.out.println("BYE BYE! Come back soon~~ YOUR WISH IS MY COMMAND <33");
+                System.out.println(line);
+                scanner.close();
+                break loop;
+            } else if (userInput.startsWith("list")) {
+                System.out.println(line);
+                for (int i = 0; i < list.size(); i++) {
+                    String numberedOutput = String.format("%d. %s", i + 1, list.get(i).toString());
+                    System.out.println(numberedOutput);
+                }
+                System.out.println(line);
+            } else {
+                try {
+                    throw new DukeException("To add a task, please start with any of these commands: 'todo', 'deadline' or 'event'!");
+                } catch (DukeException e) {
+                    System.err.println(e.getMessage());
+                    continue;
+                }
+
+            }
+
+        }
+    }
+
+    private static void saveTasks(ArrayList<Task> list) {
+        try {
+            FileWriter writer = new FileWriter(FILE_PATH);
+            for (Task task : list) {
+                writer.write(task.toFileFormat() + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            System.err.println("An error occurred while saving tasks to file.");
+            e.printStackTrace();
+        }
+    }
+
+    private static void loadTasks(ArrayList<Task> list) {
+        try {
+            File file = new File("./data/duke.txt");
+            // create parent directory if it doesn't exist.
+            if (file.getParentFile() != null && !file.getParentFile().exists()) {
+                if (!file.getParentFile().mkdirs()) {
+                    System.err.println("Failed to create parent directories.");
+                    return;
+                }
+            }
+
+            // create file if it doesn't exist.
+            if (!file.exists() && !file.createNewFile()) {
+                System.err.println("Failed to create the file.");
+                return;
+            }
+            Scanner scanner = new Scanner(file);
+            // these are for existing tasks !!
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                if (line.trim().isEmpty()) {
+                    break;
+                }
+                Task task = parseTask(line);
+                if (task != null) {
+                    list.add(task);
+                }
+            }
+            scanner.close();
+        } catch (IOException e) {
+            System.err.println("An error occurred while loading tasks from file.");
+            e.printStackTrace();
+        }
+    }
+
+    private static Task parseTask(String line) {
+        String[] parts = line.split(" \\| ");
+        Task task = null;
+
+        switch (parts[0]) {
+            case "T":
+                task = new ToDos(parts[2]);
```

Comment:
I like how you added comments to most of your switch cases, perhaps it would be good to add the case of "T" as well.